File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Features/FormComboBox.vb
Namespace CompanionKit.Controls.ComboBox.Features
	Public Class FormComboBox
		Inherits Gizmox.WebGUI.Forms.ComboBox
		' Methods
		''' <summary>
		''' Initializes a new instance of the <see cref="FormComboBox"/> class.
		''' </summary>
		''' <remarks></remarks>
		Public Sub New()
			MyBase.DropDownStyle = ComboBoxStyle.DropDown
		End Sub

		''' <summary>
		''' Gets the custom form to display as drop down
		''' </summary>
		''' <returns></returns>
		''' <remarks></remarks>
        Protected Overrides Function GetCustomDropDown() As WebGUI.Forms.Form
            Me._comboBoxForm.DialogResult = DialogResult.None
            Me._comboBoxForm.Width = Math.Max(MyBase.Width, Me._comboBoxForm.Width)
            Return Me._comboBoxForm
        End Function


		' Properties
		''' <summary>
		''' Gets a value indicating whether this instance has a custom drop down.
		''' </summary>
		''' <value><c>true</c> if this instance has a custom drop down; otherwise, <c>false</c>.</value>
		''' <returns></returns>
		''' <remarks></remarks>
		Protected Overrides ReadOnly Property IsCustomDropDown() As Boolean
			Get
				Return True
			End Get
		End Property


		' Fields
		Private _comboBoxForm As ComboBoxForm = New ComboBoxForm
	End Class
End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Features/FormComboBox.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

namespace CompanionKit.Controls.ComboBox.Features
{
    public class FormComboBox : Gizmox.WebGUI.Forms.ComboBox
    {
        private ComboBoxForm _comboBoxForm;
         /// <summary>
        /// Initializes a new instance of the <see cref="FormComboBox"/> class.
        /// </summary>
        public FormComboBox()
        {
            _comboBoxForm = new ComboBoxForm();
            this.DropDownStyle = Gizmox.WebGUI.Forms.ComboBoxStyle.DropDown;
        }

        /// <summary>
        /// Gets a value indicating whether this instance has a custom drop down.
        /// </summary>
        /// <value>
        /// 	<c>true</c> if this instance has a custom drop down; otherwise, <c>false</c>.
        /// </value>
        protected override bool IsCustomDropDown
        {
            get
            {
                return true;
            }
        }


        /// <summary>
        /// Gets the custom form to display as drop down
        /// </summary>
        /// <returns></returns>
        protected override global::Gizmox.WebGUI.Forms.Form GetCustomDropDown()
        {
            _comboBoxForm.DialogResult = Gizmox.WebGUI.Forms.DialogResult.None;
            _comboBoxForm.Width = Math.Max(this.Width, _comboBoxForm.Width);
            return _comboBoxForm;
        }
    }
}
